examples/gio_action_impl/action.rs

```diff
@@ -0,0 +1,91 @@
+use gio::{prelude::*, subclass::prelude::*};
+use glib;
+
+mod imp {
+    use super::*;
+    use std::cell::OnceCell;
+
+    #[derive(glib::Properties, Default)]
+    #[properties(wrapper_type = super::RenamedAction)]
+    pub struct RenamedAction {
+        #[property(get, construct_only)]
+        pub new_name: OnceCell<glib::GString>,
+
+        #[property(get, construct_only)]
+        pub action: OnceCell<gio::Action>,
+    }
+
+    #[glib::object_subclass]
+    impl ObjectSubclass for RenamedAction {
+        const NAME: &'static str = "ExampleRenamedAction";
+        type Type = super::RenamedAction;
+        type Interfaces = (gio::Action,);
+    }
+
+    #[glib::derived_properties]
+    impl ObjectImpl for RenamedAction {
+        fn properties() -> &'static [glib::ParamSpec] {
+            Self::derived_properties()
+        }
+
+        fn set_property(&self, id: usize, value: &glib::Value, pspec: &glib::ParamSpec) {
+            let _ = self.delegate_set_property(id, value, pspec) || {
+                self.derived_set_property(id, value, pspec);
+                true
+            };
```

Comment:
```suggestion
             if !self.delegate_set_property(id, value, pspec) {
                self.derived_set_property(id, value, pspec);
            }
```

reads less confusing

Reply:
Indeed. I just fiddled with this to see how it will work for the case of multiple interfaces. Then it would be a single expression w/o deep nesting. Anyway, here it looks ugly.